### PR TITLE
Add tests for H2GIS and PostGIS datastore

### PR DESCRIPTION
**Pull request by ebocher**
Fix also dependencies

### CODE REVIEW

File: datastore/h2gis/src/test/groovy/org/orbisgis/datastore/h2gis/H2GISTest.groovy [hunk 1]
```diff
@@ -59,4 +62,67 @@ class H2GISTest {
         assert 100 == ds.fetchSize
         assert 50 == ds.batchInsertSize
     }
+
+    @Test
+    void openH2GISFromPath() {
+        def ds = H2GIS.open("./target/db_h2gis")
+        assert ds
+    }
```
Comment (SPalominos) [inline datastore/h2gis/src/test/groovy/org/orbisgis/datastore/h2gis/H2GISTest.groovy:70]: Can you add `assert new File("./target/db_h2gis.mv.db").exists()`  to check that the db file exists.

File: datastore/h2gis/src/main/groovy/org/orbisgis/datastore/h2gis/H2GIS.groovy [hunk 2]
```diff
@@ -51,6 +51,20 @@ import groovy.transform.Field
...
+
+/**
+ * Open a H2GIS DataStore
+ * @param databasePath a path to the database
+ * @return An H2GIS DataStore.
+ */
+static JDBCDataStore open(String databasePath) {
+    Map params = new java.util.HashMap()
+    params.put("database", new File(databasePath).absolutePath)
+    params.put("dbtype", "h2gis")
```
Comment (SPalominos) [inline datastore/h2gis/src/main/groovy/org/orbisgis/datastore/h2gis/H2GIS.groovy]: This parameter can e removed as H2GISDataStoreFactory by default set `dbtype` to `h2gis`

File: datastore/h2gis/src/test/groovy/org/orbisgis/datastore/h2gis/H2GISTest.groovy [hunk 3]
```diff
@@ -59,4 +62,67 @@ class H2GISTest {
...
+                DROP TABLE IF EXISTS geotable;
+                CREATE TABLE geotable (id int, the_geom geometry(point));
+                INSERT INTO geotable VALUES (1, 'POINT(10 10)'::GEOMETRY), (2, 'POINT(1 1)'::GEOMETRY);
+        """)
+        def concat = ""
+        ds.rows "SELECT * FROM geotable", { meta ->
+            concat += "${meta.getTableName(1)} $meta.columnCount\n"
+        }
+        assertEquals("GEOTABLE 2\n", concat)
+    }
```
Comment (SPalominos) [inline datastore/h2gis/src/test/groovy/org/orbisgis/datastore/h2gis/H2GISTest.groovy]: Those tests are duplication of test written in the file [JDBCDataStoreUtilsTest](https://github.com/orbisgis/orbisgis/blob/dev/datastore/jdbc-utils/src/test/groovy/org/orbisgis/datastore/jdbcutils/JDBCDataStoreUtilsTest.groovy)

File: datastore/postgis/pom.xml [hunk 4]
```diff
@@ -54,6 +54,16 @@
...
+        <dependency>
+            <groupId>org.geotools</groupId>
+            <artifactId>gt-referencing</artifactId>
+            <version>${geotools.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.geotools</groupId>
+            <artifactId>gt-epsg-hsql</artifactId>
+            <version>${geotools.version}</version>
+        </dependency>
```
Comment (SPalominos) [inline datastore/postgis/pom.xml:66]: If those dependencies are used for tests cases they should have the tag `<scope>test</scope>`

File: datastore/h2gis/pom.xml [hunk 5]
```diff
@@ -54,6 +54,16 @@
...
+        <dependency>
+            <groupId>org.geotools</groupId>
+            <artifactId>gt-referencing</artifactId>
+            <version>${geotools.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.geotools</groupId>
+            <artifactId>gt-epsg-hsql</artifactId>
+            <version>${geotools.version}</version>
+        </dependency>
```
Comment (SPalominos) [inline datastore/h2gis/pom.xml:66]: If those dependencies are used for tests cases they should have the tag `<scope>test</scope>`

File: datastore/h2gis/src/main/groovy/org/orbisgis/datastore/h2gis/H2GIS.groovy [hunk 6]
```diff
@@ -51,6 +51,20 @@ import groovy.transform.Field
...
+    return H2GIS_DATA_STORE_FACTORY.createDataStore(params)
```
Comment (SPalominos) [inline datastore/h2gis/src/main/groovy/org/orbisgis/datastore/h2gis/H2GIS.groovy]: Can it be replaced by `return H2GIS_DATA_STORE_FACTORY.createDataStore([database:new File(databasePath).absolutePath])` ?

File: datastore/h2gis/src/test/groovy/org/orbisgis/datastore/h2gis/H2GISTest.groovy [hunk 7]
```diff
@@ -1,8 +1,11 @@
 package org.orbisgis.datastore.h2gis
 
 import org.apache.commons.dbcp.BasicDataSource
+import org.geotools.data.Query
 import org.junit.jupiter.api.Test
 
+import static org.junit.jupiter.api.Assertions.assertEquals
```
Comment (SPalominos) [inline datastore/h2gis/src/test/groovy/org/orbisgis/datastore/h2gis/H2GISTest.groovy]: `assertEquals(a, b)` can be replaced by `assert a == b`

File: datastore/h2gis/src/test/groovy/org/orbisgis/datastore/h2gis/H2GISTest.groovy [hunk 8]
```diff
@@ -59,4 +62,67 @@ class H2GISTest {
...
+    void firstRows() {
+        def ds = H2GIS.open("./target/db_h2gis")
+        assert ds
+        ds.execute("""
+                DROP TABLE IF EXISTS geotable;
+                CREATE TABLE geotable (id int, the_geom geometry(point));
+                INSERT INTO geotable VALUES (1, 'POINT(10 10)'::GEOMETRY), (2, 'POINT(1 1)'::GEOMETRY);
+        """)
+        def concat = ""
+        println ds.firstRow("select count(*) as nb from geotable").nb
```
Comment (SPalominos) [inline datastore/h2gis/src/test/groovy/org/orbisgis/datastore/h2gis/H2GISTest.groovy]: `println` here

File: datastore/h2gis/src/test/groovy/org/orbisgis/datastore/h2gis/H2GISTest.groovy [hunk 9]
```diff
@@ -59,4 +62,67 @@ class H2GISTest {
...
+        ds.execute("""
+                DROP TABLE IF EXISTS geotable;
+                CREATE TABLE geotable (id int, the_geom geometry(point));
+                INSERT INTO geotable VALUES (1, 'POINT(10 10)'::GEOMETRY), (2, 'POINT(1 1)'::GEOMETRY);
+        """)
+        def concat = ""
+        ds.rows "SELECT * FROM geotable", { meta ->
+            concat += "${meta.getTableName(1)} $meta.columnCount\n"
+        }
+        assertEquals("GEOTABLE 2\n", concat)
```
Comment (SPalominos) [inline datastore/h2gis/src/test/groovy/org/orbisgis/datastore/h2gis/H2GISTest.groovy]: May be replaced by `assert "GEOTABLE 2\n" == concat`

**Review by SPalominos — COMMENTED** [review]
Few comments.

File: datastore/postgis/pom.xml (hunk 4 above)
Comment (ebocher) [reply datastore/postgis/pom.xml:66]: You need this to work with feature

File: datastore/h2gis/src/main/groovy/org/orbisgis/datastore/h2gis/H2GIS.groovy (hunk 2 above)
Comment (ebocher) [reply datastore/h2gis/src/main/groovy/org/orbisgis/datastore/h2gis/H2GIS.groovy]: ok

**Pull request merged**